packages/pds/src/services/actor/views.ts

```diff
@@ -0,0 +1,222 @@
+        myState: {
+          follow: profileInfo?.requesterFollowing || undefined,
+          muted: !!profileInfo?.requesterMuted,
+        },
+      }
+    })
+
+    return Array.isArray(result) ? views : views[0]
+  }
+
+  profileBasic(result: UserResult, viewer: string): Promise<ProfileViewBasic>
+  profileBasic(
+    result: UserResult[],
+    viewer: string,
+  ): Promise<ProfileViewBasic[]>
+  async profileBasic(
+    result: UserResult | UserResult[],
+    viewer: string,
+  ): Promise<ProfileViewBasic | ProfileViewBasic[]> {
+    const results = Array.isArray(result) ? result : [result]
+    if (results.length === 0) return []
+
+    const { ref } = this.db.db.dynamic
+
+    const profileInfos = await this.db.db
+      .selectFrom('did_handle')
+      .where(
+        'did_handle.did',
+        'in',
+        results.map((r) => r.did),
+      )
+      .leftJoin('profile', 'profile.creator', 'did_handle.did')
+      .select([
+        'did_handle.did as did',
+        'profile.uri as profileUri',
+        'profile.displayName as displayName',
+        'profile.description as description',
+        'profile.avatarCid as avatarCid',
+        'profile.indexedAt as indexedAt',
+        this.db.db
+          .selectFrom('follow')
+          .where('creator', '=', viewer)
+          .whereRef('subjectDid', '=', ref('did_handle.did'))
+          .select('uri')
+          .as('requesterFollowing'),
+        this.db.db
+          .selectFrom('follow')
+          .whereRef('creator', '=', ref('did_handle.did'))
+          .where('subjectDid', '=', viewer)
+          .select('uri')
+          .as('requesterFollowedBy'),
+        this.db.db
+          .selectFrom('mute')
+          .whereRef('did', '=', ref('did_handle.did'))
+          .where('mutedByDid', '=', viewer)
+          .select('did')
+          .as('requesterMuted'),
+      ])
+      .execute()
+
+    const profileInfoByDid = profileInfos.reduce((acc, info) => {
+      return Object.assign(acc, { [info.did]: info })
+    }, {} as Record<string, ArrayEl<typeof profileInfos>>)
+
+    const views = results.map((result) => {
+      const profileInfo = profileInfoByDid[result.did]
+      const avatar = profileInfo?.avatarCid
+        ? this.imgUriBuilder.getCommonSignedUri('avatar', profileInfo.avatarCid)
+        : undefined
+      return {
+        did: result.did,
+        declaration: getDeclarationSimple(result),
+        handle: result.handle,
+        displayName: profileInfo?.displayName || undefined,
+        description: profileInfo?.description || undefined,
+        avatar,
+        indexedAt: profileInfo?.indexedAt || undefined,
+        viewer: {
+          muted: !!profileInfo?.requesterMuted,
+          following: profileInfo?.requesterFollowing || undefined,
+          followedBy: profileInfo?.requesterFollowedBy || undefined,
+        },
+      }
+    })
+
+    return Array.isArray(result) ? views : views[0]
+  }
+
+  // @NOTE keep in sync with feedService.getActorViews()
```

Comment:
probably worth just having one view for these (in actor service)

but don't need to fix in this PR

Reply:
💯  really wanted to do that but it got slightly dicey and I ended-up bailing on it.

Reply:
yup yup fair enough 👍